synthetics/provider.go

```diff
@@ -60,28 +44,147 @@ func NewProvider() *schema.Provider {
+			Optional:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_LOG_PAYLOADS", false),
+			Description: "Log payloads flag enables verbose debug logs of requests and responses (optional). " +
+				"Can also be specified with KTAPI_LOG_PAYLOADS environment variable.",
+		},
+	}
+}
+
+// retryProperties groups retry properties.
+// Note that, default values of schema.EnvDefaultFunc are not used, because they are not applied when user does not pass
+// retry block at all.
+func retryProperties() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		maxAttemptsKey: {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_RETRY_MAX_ATTEMPTS", nil),
+			Description: "Maximum number of request retry attempts. Default: 100. " +
+				"Can also be specified with KTAPI_RETRY_MAX_ATTEMPTS environment variable.",
+		},
+		minDelayKey: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_RETRY_MIN_DELAY", nil),
+			Description: "Minimum delay before request retry. " +
+				"Expected Go time duration format, e.g. 1s (see: <https://pkg.go.dev/time#ParseDuration>). " +
+				"Default: 1s (1 second). " +
+				"Can also be specified with KTAPI_RETRY_MIN_DELAY environment variable.",
+		},
+		maxDelayKey: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			DefaultFunc: schema.EnvDefaultFunc("KTAPI_RETRY_MAX_DELAY", nil),
+			Description: "Maximum delay before request retry. " +
+				"Expected Go time duration format, e.g. 1s (see: <https://pkg.go.dev/time#ParseDuration>). " +
+				"Default: 5m (5 minutes). " +
+				"Can also be specified with KTAPI_RETRY_MAX_DELAY environment variable.",
+		},
+	}
+}
+
 func configure(_ context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	rc, err := getRetryConfig(d)
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
+
 	cfg := kentikapi.Config{
-		SyntheticsAPIURL: getURL(d),
+		SyntheticsAPIURL: getAPIURL(d),
 		AuthEmail:        d.Get(emailKey).(string),
 		AuthToken:        d.Get(tokenKey).(string),
+		RetryCfg:         rc,
 		LogPayloads:      d.Get(logPayloadsKey).(bool),
 	}
-	log.Printf("[DEBUG] Creating Kentik API client with config: %+v", stripSensitiveData(cfg))
+
+	strippedCfg := stripSensitiveData(cfg)
+	cfgJSON, _ := json.Marshal(strippedCfg) //nolint: errcheck
+	log.Printf("Creating Kentik API client with config: %+v, JSON: %v", strippedCfg, string(cfgJSON))
 
 	return kentikapi.NewClient(cfg), nil
 }
 
-func getURL(d *schema.ResourceData) string {
-	var url string
+func getRetryConfig(d *schema.ResourceData) (kentikapi.RetryConfig, error) {
+	log.Printf("Getting retry config: %v", d.Get(retryKey))
+	retryCfg, err := getObjectFromNestedResourceData(d.Get(retryKey))
+	if err != nil {
+		return kentikapi.RetryConfig{}, fmt.Errorf("get retry configuration: %v", err)
+	}
+
+	maxAttempts, ok := retryCfg[maxAttemptsKey].(int)
+	if !ok || maxAttempts == 0 {
```

Comment:
Maybe we should log that we are backing off to default values, so it is clear what happened

Reply:
Description of properties contains info about default values. L137 logs whole config, including retry config. So we are logging all those values in a single place.

Another thing: I've noticed that user might want to disable retrying by setting maxAttempts to 0. However I cannot allow that, because in some cases SDK passes me int zero value, even if user didn't specify maxAttempts. I'll describe that case:

Having config:
```
retry {
min_delay = "100ms"
max_delay = "1m"
}
```
SDK (`d.Get(retryKey)`) passes me: `[map[max_attempts:0 max_delay:1m min_delay:100ms]]`.
This is unfortunate, in that case I need to fallback to default value.

I'll add to description that maxAttempts valid values start with 1. I think it will be good enough.